print.go

```diff
@@ -98,44 +111,43 @@ func PrintOnErrorf(format string, a ...interface{}) {
 // Spaces are added between operands when neither is a string.
 // It returns the number of bytes written and any write error encountered.
 func Fprint(writer io.Writer, a ...interface{}) {
-	if !Output {
+	pLock.Lock()
+	defer pLock.Unlock()
+	if !Output.Load() {
 		return
 	}
 
 	var ret string
 	var printed bool
 
-	for _, bar := range ActiveProgressBarPrinters {
+	activeProgressBarPrinters.lock.Lock()
+	for _, bar := range activeProgressBarPrinters.printers {
 		if bar.IsActive && bar.Writer == writer {
 			ret += sClearLine()
-			ret += Sprinto(a...)
+			ret += "\r" + color.Sprint(a...)
```

Comment:
Is there are particular reason for not using `Sprinto`?

Reply:
yeah same as previous comments, would cause a deadlock

Reply:
Could you please add that as a comment? I think that might be confusing for others at first glance. :)